template/ios/HelloWorld.xcodeproj/project.pbxproj

```diff
@@ -256,13 +256,15 @@
 			files = (
 			);
 			inputPaths = (
+				"$(SRCROOT)/.xcode.local.env",
+				"$(SRCROOT)/.xcode.env",
 			);
 			name = "Bundle React Native code and images";
 			outputPaths = (
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "set -e\n\nexport NODE_BINARY=node\n../node_modules/react-native/scripts/react-native-xcode.sh\n";
+			shellScript = "set -e\n\nif [ -f $SCRIPT_INPUT_FILE_0 ]; then\n    source $SCRIPT_INPUT_FILE_0\nelif [ -f $SCRIPT_INPUT_FILE_1 ]; then\n    source $SCRIPT_INPUT_FILE_1\nfi\n\n../node_modules/react-native/scripts/react-native-xcode.sh\n";
```

Comment:
Maybe this could be the same as the script you used above in script_phases.rb? This would prevent it from being order dependent from input files? Or we could completely hide it in a script as long as we keep the input files in the build phases?

Reply:
+1 to this. The script could just execute its parameters, e.g.:

```sh
../node_modules/react-native/scripts/with-node.sh ../node_modules/react-native/scripts/react-native-xcode.sh <args here>
```

`with-node.sh` can then be reused elsewhere without having to repeat the logic for finding the correct `.env` file to source.

Reply:
Make sense. 
I thought that leveraging the Xcode input file could be better because we are independent from the actual name of the files.. But it's probably more likely to add/remove some files as input files rather than change the `xcode.env`/`xcode.env.local` files!

Reply:
I agree that the script could lead to better code resue. Working on that as well!